packages/@ember/-internals/glimmer/lib/helpers/-resolve.ts

```diff
@@ -0,0 +1,38 @@
+/**
+@module ember
+*/
+import { Owner } from '@ember/-internals/owner';
+import { assert, runInDebug } from '@ember/debug';
+import { CapturedArguments } from '@glimmer/interfaces';
+import { createConstRef, isConstRef, valueForRef } from '@glimmer/reference';
+import { internalHelper } from './internal-helper';
+
+export default internalHelper(({ positional }: CapturedArguments, owner: Owner | undefined) => {
+  // why is this allowed to be undefined in the first place?
+  assert('[BUG] missing owner', owner);
+
+  assert(
+    `[BUG] wrong number of positional arguments, expecting 1, got ${positional.length}`,
+    positional.length === 1
+  );
+
+  let fullNameRef = positional[0];
+
+  assert('[BUG] expecting a string literal as argument', isConstRef(fullNameRef));
+
+  const fullName = valueForRef(fullNameRef);
+
+  assert('[BUG] expecting a string literal as argument', typeof fullName === 'string');
+  assert('[BUG] expecting a valid full name', fullName.split(':').length === 2);
+
+  runInDebug(() => {
```

Comment:
Seems odd to use runInDebug here (we don't do the same stripping for runInDebug as we do for the other style; ultimately we should deprecate runInDebug), can you swap to `if (DEBUG) {` instead?